network/src/testutils/builder.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::testutils::{
+    test_framework::TestFramework,
+    test_node::{NodeId, NodeType, TestNode},
+};
+use diem_config::{
+    config::NodeConfig,
+    network_id::{NetworkId, PeerNetworkId},
+};
+use rand::{rngs::StdRng, SeedableRng};
+use std::{collections::HashMap, marker::PhantomData};
+
+pub struct TestFrameworkBuilder<Node: TestNode + Sync, Framework: TestFramework<Node>> {
+    owners: u32,
```

Comment:
This is tripping me up. Is it the node count and each index is a unique id?

Reply:
It's not the node count, it's the owner count.

The idea that the owners match node owners today.